R/utils.R

```diff
@@ -734,20 +734,47 @@ read_shims <- function() {
 
 active_shims <- function() {
 
-  objects <- purrr::map(ls(name = .GlobalEnv), get)
+  text <- "get_active_shims_fertile <- function(){
```

Comment:
Huh? What is the quotation mark for?

Reply:
The quotation mark is to delineate the code for a function that's getting written to another file. Because of a weird issue with environments, inside of `active_shims()` I had to write a function to a file, source that file to get the function into the global environment, and then run it on the list of potential shims. It didn't let me just build the function in as part of `active_shims()` because when I tried that some of the base R shims were skipped over due to a difference in the environment inside the `active_shims()` function versus the global environment. It was kind of confusing, but this method does seem to give the correct results when it wasn't working before.

You can see the whole function (which I've just updated to use `map`) here. Hopefully it makes more sense like this? 

active_shims <- function() {

  text <- "is_fertile_shim <- function(i, env_obj = ls(.GlobalEnv)){

  
  object <- get(env_obj[i])


    obj_class <- class(object)[1]
    if (obj_class == 'function'){

  
      obj_attr <- attributes(object)
      if(!is.null(obj_attr$package) & !is.null(obj_attr$func_name)){
        if(obj_attr$package == 'fertile'){
          env_obj[i]
        }

      }

    }
  }"

  output_path <- fs::path(Sys.getenv("HOME"), "fertile_get_active_shims.R")
  if(file.exists(output_path)){
    fs::file_delete(output_path)
  }

  base::writeLines(text, output_path)

  base::source(output_path)

  env_obj <- ls(.GlobalEnv)

  index_vec <- 1:length(env_obj)

  is_shim_results <- as.character(purrr::map(index_vec, is_fertile_shim))

  shims_names <- is_shim_results[is_shim_results != "NULL"]

  rm("is_fertile_shim", envir = .GlobalEnv)

  shims_names

}

Reply:
Where did you get the idea for this??

Can we just write `is_fertile_shim()` to a file and include it as part of the package??

Reply:
I got the idea when I was testing an original function I wrote in fertile and found out that it worked when I copied it and executed it (added it into the global environment) then ran it, and it didn’t work when I just ran it in fertile. 

Can definitely try just adding the file to the package rather than writing it to the computer. Which folder should it go in? Inst?

Reply:
Just made another commit where I moved the file to `/inst` (next to the badge template) so that our code source files are all in the same spot and swapped the code in `active_shims()` so it sources from that file.